### PR TITLE
Implemented issue

### PR DESCRIPTION
**Pull request by dipasqualew**
Entire-Checkpoint: 05adfb6da187

### CODE REVIEW

File: apps/cli/src/lib/github.ts [hunk 1]
```diff
@@ -191,6 +191,128 @@ async function resolveReviewThread(threadId: string): Promise<boolean> {
...
+ */
+async function getVibereqReviews(prNumber: number): Promise<VibereqReview[]> {
+  const result = await runCommand([
+    "gh",
+    "api",
+    `/repos/{owner}/{repo}/pulls/${prNumber}/reviews`,
+    "--paginate",
+  ]);
+
+  if (result.exitCode !== 0) return [];
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:211]: :information_source: **getVibereqReviews filters out already-archived reviews**
Status: :white_check_mark: fulfilled

The filter correctly excludes reviews with '<details>' in their body, preventing double-archiving. Both filter conditions (has header AND no details tag) are applied correctly.

File: apps/cli/src/lib/github.ts [hunk 2]
```diff
@@ -191,6 +191,128 @@ async function resolveReviewThread(threadId: string): Promise<boolean> {
...
+        id: r.id,
+        body: r.body,
+        htmlUrl: r.html_url,
+      }));
+  } catch {
+    return [];
+  }
+}
+
+/**
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:228]: :information_source: **archiveReview wraps old review in collapsible details element**
Status: :white_check_mark: fulfilled

The collapsed body uses proper GitHub Flavoured Markdown HTML with <details>/<summary> tags on their own lines. Link to new review is included in summary.

File: apps/cli/src/lib/github.ts [hunk 3]
```diff
@@ -191,6 +191,128 @@ async function resolveReviewThread(threadId: string): Promise<boolean> {
...
+
+${originalBody}
+</details>`;
+
+  const tempFile = `/tmp/vibx-archive-${Date.now()}.json`;
+  await Bun.write(tempFile, JSON.stringify({ body: collapsedBody }));
+
+  const result = await runCommand([
+    "gh",
+    "api",
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:248]: :grey_question: **Temp file cleanup after archiveReview API call**
Status: :yellow_circle: partial

Cleanup uses 'Bun.file(tempFile).exists() && ...' which is semantically odd - exists() returns a Promise<boolean>, so the short-circuit evaluation may not work as intended. The expression evaluates the Promise object (always truthy) rather than the resolved boolean. Should use 'if (await Bun.file(tempFile).exists()) { await runCommand(["rm", tempFile]); }' instead. However, since the file was just written, it will always exist, making this a minor issue in practice.

File: apps/cli/src/lib/github.ts [hunk 4]
```diff
@@ -191,6 +191,128 @@ async function resolveReviewThread(threadId: string): Promise<boolean> {
...
+export async function archivePreviousReviews(
+  prNumber: number,
+  newReviewUrl: string,
+  logger?: Logger
+): Promise<void> {
+  const tracer = getTracer();
+
+  return withSpan(tracer, "archivePreviousReviews", async () => {
+    const previousReviews = await getVibereqReviews(prNumber);
+
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:283]: :information_source: **archivePreviousReviews resolves unresolved threads before archiving reviews**
Status: :white_check_mark: fulfilled

Threads are resolved before collapsing reviews, which is the correct order of operations.

File: apps/cli/tests/github.test.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import { REVIEW_COMMENT_HEADER } from "../src/lib/github.js";
+
+describe("REVIEW_COMMENT_HEADER", () => {
+  it("should be defined as the Vibereq review header", () => {
+    expect(REVIEW_COMMENT_HEADER).toBe("# Vibereq Review");
+  });
+});
+
+describe("archive review body formatting", () => {
+  const createArchivedBody = (originalBody: string, newReviewUrl: string): string => {
```
Comment (dipasqualew) [inline apps/cli/tests/github.test.ts:11]: :information_source: **Test coverage for archive body formatting**
Status: :white_check_mark: fulfilled

Tests verify details/summary wrapping, link inclusion, body preservation, and GFM HTML line placement. Coverage is thorough for the formatting logic.

File: apps/cli/tests/github.test.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    const originalBody = "# Vibereq Review\n\nTest";
+    const newReviewUrl = "https://github.com/owner/repo/pull/1#pullrequestreview-100";
+
+    const archived = createArchivedBody(originalBody, newReviewUrl);
+
+    // GitHub Flavoured Markdown requires HTML elements to be on their own lines
+    const lines = archived.split("\n");
+    expect(lines[0]).toBe("<details>");
+    expect(lines[lines.length - 1]).toBe("</details>");
+  });
```
Comment (dipasqualew) [inline apps/cli/tests/github.test.ts:60]: :information_source: **Test coverage for Vibereq review detection logic**
Status: :white_check_mark: fulfilled

Tests cover: header presence, no header, already-archived (has details tag), and empty/null/undefined bodies. Edge cases well covered.

**Review by dipasqualew — COMMENTED** [review]
<details>
<summary>[review superseded by <a href="https://github.com/dipasqualew/vibereq/pull/7#pullrequestreview-3862767571">new review</a>]</summary>

# Vibereq Review

## code-review
**Overall Status:** :white_check_mark: PASS

Archive previous Vibereq reviews feature implemented correctly with tests

### General Findings

### :information_source: No explicit requirements captured from intent files
**Status:** :white_check_mark: fulfilled | **Severity:** info

No intent/requirement files were provided, so analysis is based on code correctness and quality only. The implementation adds archiving of previous Vibereq reviews when submitting a new review.

### :information_source: submitReview extracts reviewUrl from API response and passes to archivePreviousReviews
**Status:** :white_check_mark: fulfilled | **Severity:** info

Review URL extraction from JSON response is properly wrapped in try/catch. archivePreviousReviews is only called if reviewUrl is defined.

### :information_source: submitReview return type updated to include optional reviewUrl
**Status:** :white_check_mark: fulfilled | **Severity:** info

Return type correctly updated to '{ success: boolean; failedFindings: Finding[]; reviewUrl?: string }' and reviewUrl is returned at end of function.

### :information_source: Dry run mode indicates archiving would occur
**Status:** :white_check_mark: fulfilled | **Severity:** info

DRY RUN path includes console.log message indicating previous reviews would be archived.

### :grey_question: No integration tests for getVibereqReviews, archiveReview, archivePreviousReviews
**Status:** :yellow_circle: partial | **Severity:** minor

Tests duplicate the filtering and formatting logic inline rather than testing the exported functions directly (only REVIEW_COMMENT_HEADER is exported and tested from the module). The core functions getVibereqReviews and archiveReview are private and untestable without exporting or refactoring. This is acceptable for now but limits test fidelity - the tests verify the logic pattern rather than the actual implementation.

</details>

File: apps/cli/src/lib/github.ts [hunk 7]
```diff
@@ -191,6 +191,128 @@ async function resolveReviewThread(threadId: string): Promise<boolean> {
...
+    if (previousReviews.length === 0) {
+      logger?.debug("No previous Vibereq reviews to archive");
+      return;
+    }
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:287]: :warning: **archivePreviousReviews resolves ALL unresolved threads (not just from previous reviews)**
Status: :yellow_circle: partial

getReviewThreads fetches all threads for the PR without filtering by review ID. This means threads from the NEW review (if any were created before archiving) could also be resolved. However, archiving happens after review creation, so any threads in the new review would be unresolved at this point - they would be incorrectly resolved. The order is: create review -> archive previous (resolves all threads) -> resolve fulfilled comments from new review. This could cause threads from the new review to be prematurely resolved before the fulfilled-comments step checks them. The fulfilled-comments resolution later re-fetches threads, so only unresolved ones remain, partially mitigating this - but the archiving step may resolve threads it shouldn't.

File: apps/cli/src/lib/github.ts [hunk 8]
```diff
@@ -343,10 +484,24 @@ export async function submitReview(
...
+
+    logger?.info("Created review", { prNumber, commentCount: comments.length, reviewUrl });
     console.log(`Created review with ${comments.length} diff comment(s)`);
 
-    // Resolve fulfilled comments
+    // Archive previous Vibereq reviews
+    if (reviewUrl) {
+      await archivePreviousReviews(prNumber, reviewUrl, logger);
+    }
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts]: :information_source: **Archive previous Vibereq reviews when submitting a new review**
Status: :white_check_mark: fulfilled

archivePreviousReviews is called after a successful review creation. The function correctly fetches all non-archived Vibereq reviews, resolves their threads, and collapses their bodies using HTML details/summary markup.

**Review by dipasqualew — COMMENTED** [review]
<details>
<summary>[review superseded by <a href="https://github.com/dipasqualew/vibereq/pull/7#pullrequestreview-3862767571">new review</a>]</summary>

# Vibereq Review

## code-review
**Overall Status:** :warning: WARN

Archive functionality is correctly implemented with good test coverage, but has edge cases around review URL extraction failure and missing dry-run coverage for archive logic

### General Findings

### :information_source: No scope creep: changes are limited to archive functionality
**Status:** :white_check_mark: fulfilled | **Severity:** info

The cleanup of the temp file deletion pattern (changing from short-circuit evaluation to an explicit if block at lines 474-478) is a minor code quality improvement that is consistent with the new archiveReview function's pattern and does not constitute problematic scope creep.

</details>

File: apps/cli/src/lib/github.ts [hunk 9]
```diff
@@ -191,8 +191,134 @@ async function resolveReviewThread(threadId: string): Promise<boolean> {
...
+    "--paginate",
+  ]);
+
+  if (result.exitCode !== 0) return [];
+
+  try {
+    const reviews = JSON.parse(result.stdout);
+    return reviews
+      .filter((r: { body: string }) => r.body?.includes(REVIEW_COMMENT_HEADER))
+      .filter((r: { body: string }) => !r.body?.includes("<details>"))
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:217]: :information_source: **Filter out already-archived reviews to avoid double-archiving**
Status: :white_check_mark: fulfilled

getVibereqReviews filters out any review whose body already contains '<details>', which is the marker used for archived reviews. This prevents re-archiving already-archived reviews.

File: apps/cli/src/lib/github.ts [hunk 10]
```diff
@@ -191,8 +191,134 @@ async function resolveReviewThread(threadId: string): Promise<boolean> {
...
+    const previousReviewIds = previousReviews.map(r => r.id);
+    const threads = await getReviewThreads(prNumber, previousReviewIds);
+
+    // Resolve threads from previous reviews only
+    for (const thread of threads) {
+      const resolved = await resolveReviewThread(thread.id);
+      if (resolved) {
+        logger?.debug("Resolved thread", { path: thread.path, line: thread.line });
+      }
+    }
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:302]: :information_source: **Resolve comment threads from previous reviews before archiving**
Status: :white_check_mark: fulfilled

getReviewThreads is called with filterReviewIds so only threads from previous Vibereq reviews are resolved, not all open threads on the PR.

File: apps/cli/src/lib/github.ts [hunk 11]
```diff
@@ -263,7 +401,7 @@ export async function submitReview(
   generalFindings: Finding[],
   dryRun = false,
   logger?: Logger
-): Promise<{ success: boolean; failedFindings: Finding[] }> {
+): Promise<{ success: boolean; failedFindings: Finding[]; reviewUrl?: string }> {
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:406]: :information_source: **Return reviewUrl from submitReview for downstream use**
Status: :white_check_mark: fulfilled

The return type of submitReview was updated to include reviewUrl?: string and the value is populated from the API response and returned.

File: apps/cli/src/lib/github.ts (hunk 8 above)
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts]: :warning: **Archive is skipped when review URL cannot be extracted**
Status: :yellow_circle: partial

When JSON parsing of the create review response fails, reviewUrl remains undefined and archivePreviousReviews is silently skipped (line 500: if (reviewUrl)). This is safe but the failure is only logged as a warning at line 493. There is no user-visible message that archiving was skipped due to missing URL, which could cause confusion when previous reviews are not archived. Consider adding a console.warn or console.log when reviewUrl is undefined to inform the operator.

File: apps/cli/src/lib/github.ts [hunk 12]
```diff
@@ -298,6 +436,7 @@ export async function submitReview(
       if (fulfilledComments.length > 0) {
         console.log(`[DRY RUN] Would resolve ${fulfilledComments.length} fulfilled comment(s)`);
       }
+      console.log(`[DRY RUN] Would archive previous Vibereq reviews`);
       return { success: true, failedFindings };
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:442]: :grey_question: **Dry-run mode should indicate archive behavior**
Status: :yellow_circle: partial

A dry-run message '[DRY RUN] Would archive previous Vibereq reviews' is added (line 439). However, the dry-run path does not actually call getVibereqReviews to show how many reviews would be archived, unlike the fulfilled-comment dry-run path which shows the count. This is a minor inconsistency in the level of detail provided in dry-run output.

File: apps/cli/src/lib/github.ts [hunk 13]
```diff
@@ -204,6 +330,13 @@ async function getReviewThreads(
               isResolved
               path
               line
+              comments(first: 1) {
+                nodes {
+                  pullRequestReview {
+                    databaseId
+                  }
+                }
+              }
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:341]: :information_source: **GraphQL query fetches review association for thread filtering**
Status: :white_check_mark: fulfilled

The GraphQL query was extended to fetch comments(first: 1) with pullRequestReview.databaseId, enabling filtering of threads by their originating review ID.

File: apps/cli/src/lib/github.ts [hunk 14]
```diff
@@ -246,6 +379,11 @@ async function getReviewThreads(
     const threads = data.data.repository.pullRequest.reviewThreads.nodes;
     return threads
       .filter((t: { isResolved: boolean }) => !t.isResolved)
+      .filter((t: { comments: { nodes: Array<{ pullRequestReview?: { databaseId?: number } }> } }) => {
+        if (!filterReviewIds || filterReviewIds.length === 0) return true;
+        const reviewId = t.comments.nodes[0]?.pullRequestReview?.databaseId;
+        return reviewId != null && filterReviewIds.includes(reviewId);
+      })
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:388]: :information_source: **Thread filter uses databaseId (integer) matching against REST API review IDs**
Status: :white_check_mark: fulfilled

The filter correctly uses databaseId (numeric) from GraphQL to match against the integer IDs returned by the REST API, which is the correct field for cross-API matching.

File: apps/cli/tests/github.test.ts [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+});
```
Comment (dipasqualew) [inline apps/cli/tests/github.test.ts:61]: :information_source: **Tests cover archive body formatting**
Status: :white_check_mark: fulfilled

Tests verify the details/summary HTML structure, the link to the new review, preservation of original body content, and GitHub Flavoured Markdown line placement requirements.

File: apps/cli/tests/github.test.ts [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+    expect(isVibereqReview(body)).toBe(false);
+  });
+
+  it("should handle empty or null bodies", () => {
+    expect(isVibereqReview("")).toBe(false);
+    // Optional chaining returns undefined for null/undefined, which is falsy
+    expect(isVibereqReview(null as unknown as string)).toBeFalsy();
+    expect(isVibereqReview(undefined as unknown as string)).toBeFalsy();
+  });
+});
```
Comment (dipasqualew) [inline apps/cli/tests/github.test.ts:96]: :information_source: **Tests cover Vibereq review detection logic**
Status: :white_check_mark: fulfilled

Tests cover: reviews with the header, reviews without the header, already-archived reviews containing <details>, and edge cases with empty/null/undefined bodies.

File: apps/cli/tests/github.test.ts [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+
+describe("archive review body formatting", () => {
+  const createArchivedBody = (originalBody: string, newReviewUrl: string): string => {
+    return `<details>
+<summary>[review superseded by <a href="${newReviewUrl}">new review</a>]</summary>
+
+${originalBody}
+</details>`;
+  };
```
Comment (dipasqualew) [inline apps/cli/tests/github.test.ts:17]: :grey_question: **Tests are unit tests that do not require external process calls**
Status: :yellow_circle: partial

The test file duplicates the createArchivedBody and isVibereqReview logic inline rather than testing the actual exported functions (archiveReview and getVibereqReviews are not exported and cannot be tested directly). This means the tests verify the formatting logic independently but do not guard against divergence between the test helper and the actual implementation in archiveReview at github.ts line 237. If the template in archiveReview changes, the tests will not catch it unless the test helper is also updated. Consider exporting the formatting functions or testing through integration.

File: apps/cli/src/lib/github.ts [hunk 18]
```diff
@@ -191,8 +191,134 @@ async function resolveReviewThread(threadId: string): Promise<boolean> {
...
+/**
+ * Get all Vibereq reviews on a PR (identified by the REVIEW_COMMENT_HEADER)
+ */
+async function getVibereqReviews(prNumber: number): Promise<VibereqReview[]> {
+  const result = await runCommand([
+    "gh",
+    "api",
+    `/repos/{owner}/{repo}/pulls/${prNumber}/reviews`,
+    "--paginate",
+  ]);
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts:209]: :information_source: **Pagination handled when fetching previous reviews**
Status: :white_check_mark: fulfilled

The gh api call for fetching PR reviews uses --paginate, ensuring all reviews are retrieved even on PRs with many reviews.

File: apps/cli/src/lib/github.ts [hunk 19]
```diff
@@ -191,8 +191,134 @@ async function resolveReviewThread(threadId: string): Promise<boolean> {
...
+ */
+export async function archivePreviousReviews(
+  prNumber: number,
+  newReviewUrl: string,
+  logger?: Logger
+): Promise<void> {
+  const tracer = getTracer();
+
+  return withSpan(tracer, "archivePreviousReviews", async () => {
+    const previousReviews = await getVibereqReviews(prNumber);
```
Comment (dipasqualew) [inline apps/cli/src/lib/github.ts]: :warning: **Race condition: new review is included in 'previous reviews' list**
Status: :x: violation

archivePreviousReviews calls getVibereqReviews after the new review has already been created (line 501 calls archivePreviousReviews after the POST succeeds). The new review body starts with REVIEW_COMMENT_HEADER and does not contain '<details>', so it will pass the filter in getVibereqReviews and may be included in previousReviews. This would cause the new review to be immediately archived. The fix is to either pass the new review's ID to archivePreviousReviews and exclude it, or rely on the fact that the new review's html_url is used as the archive link (a review cannot link to itself). However, since the filtering only checks for the header and absence of '<details>', the new review is at risk of being archived in the same operation that created it.
